### PR TITLE
Jm/season stats 1

### PR DESCRIPTION
**Pull request by jdmchugh111**
I created tests and wrote the code for all of the Season Statistics. I had to make changes to stat_tracker.rb and game_teams.rb in order to incorporate game_id as an attribute.

### CODE REVIEW

File: lib/game_team.rb [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
+                :team_id,
                 :home_or_away,
                 :result,
                 :head_coach,
                 :goals,
                 :shots,
                 :tackles
 
-    def initialize(team_id, home_or_away, result, head_coach, goals, shots, tackles)
+    def initialize(game_id, team_id, home_or_away, result, head_coach, goals, shots, tackles)
```
Comment (cheeleertr) [inline lib/game_team.rb:13]: Good to have game_id

File: lib/season_stats.rb [hunk 2]
```diff
@@ -1,4 +1,166 @@
+module SeasonStats
+    
+    def winningest_coach(season_id)
```
Comment (cheeleertr) [inline lib/season_stats.rb:3]: Nice use of helper methods to make this clean.

File: lib/stat_tracker.rb
Comment (cheeleertr) [inline lib/stat_tracker.rb:1]: Good catch on the uppercase to lowercase change from the header_converters.

File: spec/spec_helper.rb
Comment (cheeleertr) [inline spec/spec_helper.rb:1]: We can work on getting simplecov to work later.

File: lib/season_stats.rb
Comment (cheeleertr) [inline lib/season_stats.rb:1]: Great job on the methods. Nice and clean. You're a beast getting these done in 1 night.

Comment (JCL461437) [reply lib/season_stats.rb:1]: ^^^ James is a beast!

File: lib/game_team.rb (hunk 1 above)
Comment (JCL461437) [reply lib/game_team.rb:13]: Yes, definitely

**Review by JCL461437 — APPROVED** [review]
Overall, looks great man! Very impressive.

**Pull request merged**